apps/coscrad-frontend/src/utils/generic-components/presenters/tables/generic-index-table-presenter/index-table.tsx

```diff
@@ -196,14 +201,47 @@ export const IndexTable = <T extends IBaseViewModel>({
     return (
         <div>
             <h3>{heading}</h3>
+
+            <FormControl sx={{ minWidth: 120 }} size={'small'}>
+                <InputLabel>Filter</InputLabel>
+                <Select
+                    label={'Filter'}
+                    value={selectedFilterProperty}
+                    onChange={(changeEvent) => {
+                        const {
+                            target: { value },
+                        } = changeEvent;
+                        setSelectedFilterProperty(value as keyof T);
+                    }}
+                >
+                    <MenuItem sx={{ minWidth: 120 }} value={'allColumns'}>ALL</MenuItem>
```

Comment:
You need to declare a constant at the top 
```ts
const allColumns = `allColumns`
// or better const allColumns = `*`
```
because this magic string is used more than once. It's also a good candidate for configurable text, but don't worry about that right now.